course/views.py

```diff
@@ -37,29 +37,33 @@ def get_common_objects(self):
         self.all_instances = []
         end_threshold = timezone.now() - datetime.timedelta(days=30)
         user = self.request.user
+        my_instances = []
 
         if user and user.is_authenticated:
-            for course in user.userprofile.teaching_courses.all():
-                for instance in course.instances.all().filter(ending_time__gte=end_threshold):
-                    self.my_instances.append(instance)
+            for instance in CourseInstance.objects.filter(course__teachers=user.userprofile)\
+                .filter(ending_time__gte=end_threshold):
```

Comment:
Typically, one should avout `\` when possible. One good place is `with` statement, where there is no better alternatives. Bot for here, you could consider formatting this like (key idea is to use brackets, mutliple ideas how to format it though):

```python
for instance in (CourseInstance.objects
                 .filter(course__teachers=user.userprofile,
                         ending_time__gte=end_threshold)
                 .all()):
    my_instances.append(instance)
```

or

```python
for instance in (CourseInstance.objects
                 .filter(
                     course__teachers=user.userprofile,
                     ending_time__gte=end_threshold,
                 )
                 .all()):
    my_instances.append(instance)
```

or

```python
my_instances.extend(
    CourseInstance.objects
    .filter(
        course__teachers=user.userprofile,
        ending_time__gte=end_threshold,
    )
    .all()
)
```

I guess the `.all()` could be left from the latter two, as it's not important to ensure that the Django won't use a cached queryset. In the first, it helps readability.

Reply:
This was also now fixed for the query with multiple filters. So all yesterday's review comments should have now been taken into consideration.